### PR TITLE
Use username for WP.com login in Jetpack SSO test

### PR DESCRIPTION
**Pull request by brbrr**
This PR fixes login limit error by switching from email to username login. Also, some refactoring is included.

### CODE REVIEW

**Comment by brbrr** [comment]
Test failures are not related to Jetpack Connect via SSO test. So it should be good to merge once approved.

**Comment by alisterscott** [comment]
The prettier formatting makes it really hard to work out what has changed versus what has just been reformatted 😞

**Comment by alisterscott** [comment]
Works as expected:

```
env BROWSERSIZE=desktop ./node_modules/.bin/mocha specs-jetpack-calypso


  Jetpack Connect: (desktop)
    Connect via SSO: @parallel @jetpack
Found an AB Testing key in local storage that isn't known: 'showMoneyBackGuarantee'. This may cause inconsistent A/B test behaviour, please check this is okay and add it to 'knownABTestKeys' in default.config
      ✓ Can register new Subscriber user (27192ms)
      ✓ Can log into WordPress.com (9118ms)
      ✓ Can log into site via Jetpack SSO (9718ms)
      ✓ Add new user as Subscriber in wp-admin (5491ms)
      ✓ Log out from WP Admin (3994ms)
      ✓ Can log in as Subscriber (6836ms)
The localstorage value for AB tests wasn't set correctly.
Expected value is:
'{"upgradePricingDisplayV2_20180305":"original","skipThemesSelectionModal_20170904":"show","checklistThankYouForFreeUser_20171204":"hide","checklistThankYouForPaidUser_20171204":"hide","redesignedSidebarBanner_20180222":"oldBanner","businessPlanDescriptionAT_20170605":"original","mobilePlansTablesOnSignup_20180330":"original"}'
Actual value is:
'{"upgradePricingDisplayV2_20180305":"original","skipThemesSelectionModal_20170904":"show","checklistThankYouForFreeUser_20171204":"hide","checklistThankYouForPaidUser_20171204":"hide","redesignedSidebarBanner_20180222":"oldBanner","businessPlanDescriptionAT_20170605":"original","mobilePlansTablesOnSignup_20180330":"original","showMoneyBackGuarantee_20180409":"no"}'
      ✓ Can login via SSO into WP Admin (14509ms)


  7 passing (1m)
```

File: specs-jetpack-calypso/wp-jetpack-connect-spec.js [hunk 1]
```diff
@@ -274,61 +275,52 @@ test.describe( `Jetpack Connect: (${ screenSize })`, function() {
...
-			this.signupFlow = new SignUpFlow( driver, {
+
+			const signupFlow = new SignUpFlow( driver, {
 				emailAddress: this.emailAddress,
+				blogName: this.blogName,
 				password: this.password,
 			} );
-			return this.signupFlow
+			return signupFlow
 				.signupFreePlan()
```
Comment (alisterscott) [inline specs-jetpack-calypso/wp-jetpack-connect-spec.js]: Not really relevant to this PR but we don't need to sign up for a free site, we could just create a user account using https://wordpress.com/start/account

**Comment by alisterscott** [comment]
I updated the SSO spec to just create an account for the customer in this PR: https://github.com/Automattic/wp-e2e-tests/pull/1144

**Pull request merged**